lib/ruby_llm/providers/bedrock/streaming/message_processing.rb

```diff
@@ -56,9 +56,110 @@ def valid_payload?(payload)
             json_start = payload.index('{')
             json_end = payload.rindex('}')
 
-            return false if json_start.nil? || json_end.nil? || json_start >= json_end
+            !(json_start.nil? || json_end.nil? || json_start >= json_end)
+          end
+
+          def extract_headers(chunk, headers_start, headers_end)
+            headers_data = chunk[headers_start...headers_end]
+            parse_headers(headers_data)
+          end
+
+          def parse_headers(headers_data)
```

Comment:
I'd appreciate a comment on such a long method name. Why do we need such monstrosity?

Reply:
@crmne I noticed in another PR you mentioned not wanting any dependencies on AWS SDKs. 

Would you accept this gem https://rubygems.org/gems/aws-eventstream/versions/1.0.0? Using this would remove the need for this crazy stream header parsing. 

If not then thats fine, I can clean up my existing implementation. I went with the approach of first make it work, then make it pretty.